test/test_quantized_models.py

```diff
@@ -1,93 +1,53 @@
-        x = torch.rand(input_shape)
-        model(x)
-        return
-
-    def check_script(self, model, name):
-        if name in scriptable_quantizable_models_blacklist:
-            return
-        scriptable = True
-        msg = ""
-        try:
-            torch.jit.script(model)
-        except Exception as e:
-            tb = traceback.format_exc()
-            scriptable = False
-            msg = str(e) + str(tb)
-        self.assertTrue(scriptable, msg)
-
-    def _test_classification_model(self, name, input_shape):
-        # First check if quantize=True provides models that can run with input data
-
-        model = torchvision.models.quantization.__dict__[name](pretrained=False, quantize=True)
-        self.check_quantized_model(model, input_shape)
-
-        for eval_mode in [True, False]:
-            model = torchvision.models.quantization.__dict__[name](pretrained=False, quantize=False)
-            if eval_mode:
-                model.eval()
-                model.qconfig = torch.quantization.default_qconfig
-            else:
-                model.train()
-                model.qconfig = torch.quantization.default_qat_qconfig
-
-            model.fuse_model()
-            if eval_mode:
-                torch.quantization.prepare(model, inplace=True)
-            else:
-                torch.quantization.prepare_qat(model, inplace=True)
-                model.eval()
-
-            torch.quantization.convert(model, inplace=True)
-
-        self.check_script(model, name)
-
-
-for model_name in get_available_quantizable_models():
-    # for-loop bodies don't define scopes, so we have to save the variables
-    # we want to close over in some way
-    def do_test(self, model_name=model_name):
-        input_shape = (1, 3, 224, 224)
-        if model_name in ['inception_v3']:
-            input_shape = (1, 3, 299, 299)
-        self._test_classification_model(model_name, input_shape)
-
-    # inception_v3 was causing timeouts on circleci
-    # See https://github.com/pytorch/vision/issues/1857
-    if model_name not in ['inception_v3']:
-        setattr(ModelTester, "test_" + model_name, do_test)
+@pytest.mark.skipif(not ('fbgemm' in torch.backends.quantized.supported_engines and
+                         'qnnpack' in torch.backends.quantized.supported_engines),
+                    reason="This Pytorch Build has not been built with fbgemm and qnnpack")
+# inception_v3 was causing timeouts on circleci
+# See https://github.com/pytorch/vision/issues/1857
+@pytest.mark.parametrize('name', [name for name in get_available_quantizable_models() if name != 'inception_v3'])
+def test_classification_model(name):
+
+    # First check if quantize=True provides models that can run with input data
+    input_shape = (1, 3, 224, 224)
+    model = torchvision.models.quantization.__dict__[name](pretrained=False, quantize=True)
+    x = torch.rand(input_shape)
+    model(x)
+
+    for eval_mode in [True, False]:
```

Comment:
Just question: How come you did not parameterize this one?

Reply:
because TBH I don't know what this part is about. I'm not sure whether it is related to the scripted part below or not, so I prefered not to touch it.